embedded/store/tx.go

```diff
@@ -107,11 +86,22 @@ func (hdr *TxHeader) Bytes() []byte {
 	binary.BigEndian.PutUint64(b[i:], uint64(hdr.Ts))
 	i += tsSize
 
-	binary.BigEndian.PutUint16(b[i:], uint16(len(mdbs)))
+	binary.BigEndian.PutUint16(b[i:], uint16(hdr.Version))
 	i += sszSize
 
-	copy(b[i:], mdbs)
-	i += len(mdbs)
+	if hdr.Version > 0 {
+		var mdbs []byte
+
+		if hdr.Metadata != nil {
+			mdbs = hdr.Metadata.Bytes()
+		}
+
+		binary.BigEndian.PutUint16(b[i:], uint16(len(mdbs)))
+		i += sszSize
+
+		copy(b[i:], mdbs)
+		i += len(mdbs)
+	}
 
 	binary.BigEndian.PutUint16(b[i:], uint16(hdr.NEntries))
```

Comment:
As in case of ImmuStore.performCommit, to consider whether we shouldn't store nentries in 32-bit integer in case of a v1 encoding

Reply:
lets have a discussion about big transactions, use cases and scalability, related to the discussion in the comments above